observatory-platform/observatory/platform/terraform/api/main.tf

```
@@ -0,0 +1,163 @@
+        }
+        env {
+          name = "ES_HOST"
+          value = "sm://${var.google_cloud.project_id}/elasticsearch-host"
+        }
+      }
+      service_account_name = google_service_account.api-backend_service_account.email
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+  depends_on = [module.elasticsearch-logins]
+}
+
+########################################################################################################################
+# Cloud Endpoints API
+########################################################################################################################
+# Random id to use in endpoints hostname
+resource "random_id" "endpoint-suffix" {
+  byte_length = 8
+}
+
+locals {
+  endpoints_host = "api-${random_id.endpoint-suffix.hex}.endpoints.${var.google_cloud.project_id}.cloud.goog"
+}
+
+# Create/update endpoints configuration based on OpenAPI
+resource "google_endpoints_service" "api" {
+  project = var.google_cloud.project_id
+  service_name = local.endpoints_host
+  openapi_config = templatefile("./openapi_endpoint.yml", {
+    host = local.endpoints_host,
+    backend_address = google_cloud_run_service.backend.status[0].url,
+    query_parameters = ["id", "name", "published_year", "coordinates", "country", "country_code", "region",
+                               "subregion", "access_type", "label", "status", "collaborator_coordinates",
+                               "collaborator_country", "collaborator_country_code", "collaborator_id",
+                               "collaborator_name", "collaborator_region", "collaborator_subregion", "field", "source",
+                               "funder_country_code", "funder_name", "funder_sub_type", "funder_type", "journal",
+                               "output_type", "publisher"]
+  })
+}
+
+# Create service account used by Cloud Run
+resource "google_service_account" "api-endpoint_service_account" {
+  account_id = "api-endpoint"
+  display_name = "API endpoint Service Account"
+  description = "The Google Service Account used by the cloud run endpoint"
+}
+
+
+# Give permission to Cloud Run Endpoint service-account to access private Cloud Run backend
+resource "google_project_iam_member" "api-endpoint_service_account_cloudrun_iam" {
+  project = var.google_cloud.project_id
+  role    = "roles/run.invoker"
+  member  = "serviceAccount:${google_service_account.api-endpoint_service_account.email}"
+}
+
+# Give permission to Cloud Run Endpoint service-account to control service management
+resource "google_project_iam_member" "api-endpoint_service_account_servicecontroller_iam" {
+  project = var.google_cloud.project_id
+  role    = "roles/servicemanagement.serviceController"
+  member  = "serviceAccount:${google_service_account.api-endpoint_service_account.email}"
+}
+
+# Create/update Cloud Run service
+resource "google_cloud_run_service" "endpoints-cloud-run" {
```

```suggestion
resource "google_cloud_run_service" "api_endpoints" {
```
It would be good to make the naming of this consistent with the other cloud run service. So perhaps call this one api_endpoints and the backend api_backend?